### PR TITLE
Feature/rails3 fix watcher routes

### PR DESCRIPTION
**Pull request by tessi**
create a pull request for @ulferts 
this fixes the watcher routes for rails3 and makes them testable

### CODE REVIEW

**Comment by tessi** [comment]
I get a warning when running the specs
<code><pre>WARNING: Shared example group 'watched model' has been previously defined at:
  /home/philipp/finn/openproject_core/openproject/spec/lib/acts_as_watchable/lib/acts_as_watchable/routes_spec.rb:12
...and you are now defining it at:
  /home/philipp/finn/openproject_core/openproject/spec/routing/watchers_spec.rb:4
The new definition will overwrite the original one.
</pre></code>

File: spec/lib/acts_as_watchable/lib/acts_as_watchable/routes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      describe "for an invalid id string" do
+        let(:id) { "schmu" }
+
+        it "should not be false" do
+          OpenProject::Acts::Watchable::Routes.matches?(request).should be_false
+        end
+      end
+    end
+
+    ['issues', 'news', 'news', 'boards', 'messages', 'wikis', 'wiki_pages'].each do |type|
```
Comment (tessi) [inline spec/lib/acts_as_watchable/lib/acts_as_watchable/routes_spec.rb:31]: you have two 'news' string listed in that array. propably only one is needed.

**Comment by tessi** [comment]
I know how to tests issue-watching. But what about news, boards, messages, wikis, wiki_pages. How can I test watching them through the openproject UI. Does someone know?

**Comment by tessi** [comment]
Apart from that I let the tests run on my system and they seem to work. Issue-watching works too (watch, unwacht, mails are sent when appropriate etc.).

**Comment by ulferts** [comment]
I have fixed the offending points in routes/watchers_spec.rb in  bc7292b 

There currently is no functionality on the UI level for watching news, wikis and wiki_pages. There is for boards and messages however (albeit a bug in forum#create prevents creating a forum for now). I have checked it for forums and messages (by fixing the bug with duct-tape) and it works for me.

Albeit the functionality not beeing accessible for news, wikis and wiki_pages. I would still recommend leaving the routes as it seems to me that the missing UI support is the bug and not the routes.

**Pull request merged**